image_transport_py/src/image_transport_py/pybind_image_transport.cpp

```diff
@@ -0,0 +1,212 @@
+    "Unsubscribe the callback associated with this CameraPublisher.")
+  .def(
+    "publish",
+    [](image_transport::CameraPublisher & publisher, sensor_msgs::msg::Image & img,
+    sensor_msgs::msg::CameraInfo & info) {
+      publisher.publish(img, info);
+    },
+    "Publish an image and camera info on the topics associated with this Publisher.");
+
+  pybind11::class_<image_transport::ImageTransport>(m, "ImageTransport")
+  .def(
+    pybind11::init(
+      [](const std::string & node_name, const std::string & image_transport,
+      const std::string & launch_params_filepath) {
+        if (!rclcpp::ok()) {
+          rclcpp::init(0, nullptr);
+        }
+        rclcpp::NodeOptions node_options;
+
+        if (!launch_params_filepath.empty()) {
+          node_options.allow_undeclared_parameters(true)
+          .automatically_declare_parameters_from_overrides(true)
+          .arguments({"--ros-args", "--params-file", launch_params_filepath});
+        } else if (!image_transport.empty()) {
+          node_options.allow_undeclared_parameters(true)
+          .automatically_declare_parameters_from_overrides(true)
+          .append_parameter_override("image_transport", image_transport);
+        }
+
+        rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared(node_name, "", node_options);
+
+        std::shared_ptr<rclcpp::executors::MultiThreadedExecutor> executor =
+        std::make_shared<rclcpp::executors::MultiThreadedExecutor>();
+
+        auto spin_node = [node, executor]() {
+          executor->add_node(node);
+          executor->spin();
+        };
+        std::thread execution_thread(spin_node);
+        execution_thread.detach();
+
+        return image_transport::ImageTransport(node);
+      }),
+    pybind11::arg("node_name"), pybind11::arg("image_transport") = "",
+    pybind11::arg("launch_params_filepath") = "",
+
+    "Initialize an ImageTransport object with a node name, image_transport"
+    " and launch params file path.")
+  .def(
+    "advertise",
+    pybind11::overload_cast<const std::string &, uint32_t, bool>(
+      &image_transport::ImageTransport::advertise),
+    pybind11::arg("base_topic"), pybind11::arg("queue_size"), pybind11::arg("latch") = false,
+    "Advertise an image topic.")
+  .def(
+    "advertise_camera",
+    pybind11::overload_cast<const std::string &, uint32_t, bool>(
+      &image_transport::ImageTransport::advertiseCamera),
+    pybind11::arg("base_topic"), pybind11::arg("queue_size"), pybind11::arg("latch") = false,
+    "Advertise an image topic with camera info.")
+  .def(
+    "subscribe",
+    [](image_transport::ImageTransport & image_transport,
+    const std::string & base_topic,
+    uint32_t queue_size,
+    const image_transport::Subscriber::Callback & callback) {
+
+      // Static vector to keep the subscriptions alive
+      thread_local auto vec = std::vector<std::shared_ptr<image_transport::Subscriber>>();
```

Comment:
I did not find a better and safer way to make `Subscriber` available for multiple threads from Python and C++. The `Subscriber` can be shut down, but the memory resource will only be freed when the `rclpy` node's thread exists. I think it is a reasonable tradeoff compared to making changes in `image_transport` to store state and/or go deep into python/c++ object ownership management.